src/modules/hip/kernel/slice.hpp

```diff
@@ -32,77 +87,441 @@ __global__ void slice_ncdhw_hip_tensor(T *srcPtr,
+
+            // if needPadding is set, launch kernel for filling the padded region with fill value specified
+            if (needPadding && dstGenericDescPtr->layout == RpptLayout::NCDHW)
+            {
+                hipLaunchKernelGGL(fill_value_ncdhw_hip_tensor,
+                                   dim3(ceil((float)globalThreads_x/LOCAL_THREADS_X), ceil((float)globalThreads_y/LOCAL_THREADS_Y), ceil((float)globalThreads_z/LOCAL_THREADS_Z)),
+                                   dim3(LOCAL_THREADS_X, LOCAL_THREADS_Y, LOCAL_THREADS_Z),
+                                   0,
+                                   handle.GetStream(),
+                                   dstPtr + (batchCount * dstGenericDescPtr->strides[0]),
+                                   make_uint3(dstGenericDescPtr->strides[1], dstGenericDescPtr->strides[2], dstGenericDescPtr->strides[3]),
+                                   dstGenericDescPtr->dims[1],
+                                   make_uint3(maxDepth, maxHeight, maxWidth),
+                                   fillValue);
+            }
+            else if (needPadding && dstGenericDescPtr->layout == RpptLayout::NDHWC)
+            {
+                hipLaunchKernelGGL(fill_value_ndhwc_hip_tensor,
+                                   dim3(ceil((float)globalThreads_x/LOCAL_THREADS_X), ceil((float)globalThreads_y/LOCAL_THREADS_Y), ceil((float)globalThreads_z/LOCAL_THREADS_Z)),
+                                   dim3(LOCAL_THREADS_X, LOCAL_THREADS_Y, LOCAL_THREADS_Z),
+                                   0,
+                                   handle.GetStream(),
+                                   dstPtr + (batchCount * dstGenericDescPtr->strides[0]),
+                                   make_uint2(dstGenericDescPtr->strides[1], dstGenericDescPtr->strides[2]),
+                                   make_uint3(maxDepth, maxHeight, maxWidth),
+                                   fillValue);
+            }
+        }
+    }
+    else if(numDims == 3)
+    {
+        // set the dimsOrder and globalthreads values required for NHWC layout
+        Rpp32s dimsOrder[2] = {0, 1};
+        int globalThreads_x = (dstGenericDescPtr->strides[1] / 3 + 7) >> 3; // W - width  (x direction) - vectorized for 8 element loads/stores per channel
+        int globalThreads_y = dstGenericDescPtr->dims[1];                   // H - height (y direction)
+        int globalThreads_z = 1;
+
+        // change the dimsOrder and globalthreads values if layout is NCHW
+        if((srcGenericDescPtr->layout == RpptLayout::NCHW) && (dstGenericDescPtr->layout == RpptLayout::NCHW))
+        {
+            dimsOrder[0] = 1;  // height
+            dimsOrder[1] = 2;  // width
+            int globalThreads_x = (dstGenericDescPtr->strides[2] + 7) >> 3; // W - width  (x direction) - vectorized for 8 element loads/stores per channel
+            int globalThreads_y = dstGenericDescPtr->dims[2];               // H - height (y direction)
```

Comment:
There is something definitely wrong here if it works with this, since scope of the variable is within the if

Reply:
Okay the numDims = 4 is correct.
Just for numDims = 3 please remove the re-declaration of int for L200-202

Reply:
Not sure how this kernel was working with global x and global y using wrong values